Added Getting started section (#153)
etn-ccis/blui-react-native: docs/src/componentDocs/GettingStarted/ReactNative.mdx

import React from 'react';
import { Divider } from '../../layout/Divider';
import { CodeBlock } from '../../shared';

> {/* KEYWORDS: react web framework guide tutorial */}

Reviewer: ```suggestion
{/* KEYWORDS: react react native framework templates guide tutorial */}
```

# Getting Started with Brightlayer UI

> The following instructions will guide you through starting a new Brightlayer UI project using React. Alternatively, you can also start your project by downloading one of our code samples from [Github](https://github.com/etn-ccis?q=blui) (the Login or Navigation examples are good projects to start from).

Reviewer: ```suggestion
The following instructions will guide you through starting a new Brightlayer UI project using React Native. Alternatively, you can also start your project by downloading one of our code samples from [Github](https://github.com/etn-ccis?q=blui) (the Login or Navigation examples are good projects to start from).
```

# Generating a New Project with the Brightlayer UI React Native Templates

The fastest way to start a new Brightlayer UI project is to use the [Brightlayer UI React Native Templates](https://brightlayer-ui.github.io/resources/developer). This will automatically create a starter project for you with all of the necessary Brightlayer UI dependencies pre-configured. To start a React Native project use any of these 3 different configurations:

<CodeBlock code={`npx @react-native-community/cli init MyApp --template @brightlayer-ui/react-native-template-blank-typescript
npx @react-native-community/cli init MyApp --template @brightlayer-ui/react-native-template-routing-typescript
npx @react-native-community/cli init MyApp --template @brightlayer-ui/react-native-template-authentication-typescript`} language={'sh'} />

You will be asked to provide a name for your project and whether you want to use the Brightlayer UI eslint and prettier configuration files.

> The Brightlayer UI CLI will automatically include:

Reviewer: ```suggestion
The Brightlayer UI templates will automatically include:
```

## Adding Brightlayer UI Themes

By default, your application will be themed to use the [Brightlayer UI theme](https://www.npmjs.com/package/@brightlayer-ui/react-native-themes).

If you are using the recommended component libraries, your application will be themed with the Google Material theme by default. To use the Brightlayer UI theme, first install it:

<CodeBlock code={`npm install --save @brightlayer-ui/react-native-themes`} language={'sh'} />
or
<CodeBlock code={`yarn add @brightlayer-ui/react-native-themes`} language={'sh'} />

> Using the Brightlayer UI React Native theme requires that you add the Open Sans font to your application. You can learn how to do this by reading the instructions for Vanilla React Native. You can find the fonts here.

### Applying the Theme
Import the theme files and React Native Paper theme provider into your application, e.g. in App.js:

<CodeBlock code={`import { Provider as ThemeProvider } from 'react-native-paper';
import * as BLUIThemes from '@brightlayer-ui/react-native-themes';
...
<ThemeProvider theme={BLUIThemes.blue}>
    <App />
</ThemeProvider>`} language={'sh'} />

> For detailed information about theming, refer to the react-native-paper theming guide.

Reviewer: can we link to this theme guide

Author: link added for theming guide

> NOTE: This theme only applies to the components in the react-native-paper and @brightlayer-ui/react-native-components package. It will not theme the components from other third party libraries. If you wish to use our theme to style your own components, refer to the react-native-paper theming guide.

<Divider />

# Writing a React Native Application
If you are familiar with React development, React Native will be relatively simple to understand. The application logic remains the same, while the UI requires a little adjustment. Rather than typical HTML elements, React Native UIs are built with a special library of Components, including `<View>, <Text>, <Image>`, etc. React Native layouts are built exclusively using FlexBox.

## Running your application

> After you've created your application with a CLI, it can be run from the command line:

Reviewer: ```suggestion
After you've created your application using the Brightlayer UI templates, it can be run from the command line:
```
